src/lib.rs

```diff
@@ -56,17 +56,18 @@ impl Diagnostic {
     }
 
     pub fn unexpected_token_found(
-        expected: String,
+        expected: Option<String>,
```

Comment:
I was thinking if we really need the option here.
Since the vast majority of the calls actually give an expected value, I thought we could try and find one for the cases where it's still unclear.
They are usually cases where we have a big match branch, but I think they have something in common usually.
I am commenting the cases as I see them.